core/state/state_object.go

```diff
@@ -528,14 +514,10 @@ func (s *stateObject) CodeSize() int {
 }
 
 func (s *stateObject) SetCode(codeHash common.Hash, code []byte) {
-	prevcode := s.Code()
-	s.db.journal.append(codeChange{
-		account:  &s.address,
-		prevhash: s.CodeHash(),
-		prevcode: prevcode,
-	})
+	s.db.journal.JournalSetCode(s.address)
```

Comment:
What about changing the semantics a bit. As ethereum contract code can't be changed unless: (a) creation, new code (b) destruct, empty code. Otherwise it's not supported to upgrade code from code_a to code_b.

Although from the statedb's perspective, it's totally possible to support code upgrade. Do you think it makes sense to be a bit more strict here?

e.g. if len(s.Code()) > 0; then reject. 

It will align with the journal's behavior.

Reply:
I'm not sure what you are proposing. Please clarify a bit

Reply:
The SetCode method doesn't return any error , so rejecting is a bit problematic